### PR TITLE
Theme Generator tool

### PR DESCRIPTION
**Pull request by phkuo**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

The Theme Generator is a tool to help users easily create themes usable with our styling package. The user selects a few colors and the tool automatically fills in the Fabric palette slots to create a complete theme. (Hidden functionality is also included to generate a theme from an image. It requires the user create a local enlistment and adding their own subscription key.)

image

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by phkuo** [comment]
redo of #2881 due to #2913

File: apps/fabric-website/src/components/App/AppState.tsx [hunk 1]
```diff
@@ -380,6 +385,7 @@ export const AppState: IAppState = {
             {
               title: 'Themes',
               url: '#/components/themes',
+              isHiddenFromMainNav: true,
```
Comment (Jahnp) [inline apps/fabric-website/src/components/App/AppState.tsx]: I don't think we'd want to hide this right now, do we? This functionality is useful to demo, even if the generator tool isn't available.

File: apps/fabric-website/src/components/App/AppState.tsx [hunk 2]
```diff
@@ -74,6 +74,11 @@ export const AppState: IAppState = {
           getComponent: cb => require.ensure([], (require) => cb(require<any>('../../pages/Styles/LocalizationPage/LocalizationPage').LocalizationPage))
         },
         {
+          title: 'Beta Theme Generator',
+          url: '#/styles/themeGenerator',
+          getComponent: cb => require.ensure([], (require) => cb(require<any>('../../pages/Styles/ThemeGeneratorToolPage/ThemeGeneratorToolPage').ThemeGeneratorToolPage))
+        },
```
Comment (lynamemi) [inline apps/fabric-website/src/components/App/AppState.tsx]: @Jahnp this is what needs to be hidden, right?

Comment (Jahnp) [reply apps/fabric-website/src/components/App/AppState.tsx]: Yep, exactly. @phkuo--this is the link you want to add `isHiddenFromMainNav: true` to.

**Pull request merged**